spec/classes/puppet_server_puppetserver_spec.rb

```diff
@@ -404,6 +404,40 @@
         end
       end
 
+      describe 'server_max_open_files', unless: facts[:osfamily] == 'FreeBSD' do
+        context 'when server_max_open_files => undef' do
+          it do
+            should contain_file_line('puppet::server::puppetserver::max_open_files')
+              .with_ensure('absent')
+          end
+        end
+
+        context 'when server_max_open_files => 32143' do
+          let(:params) { super().merge(server_max_open_files: 32143) }
+
+          context 'on systemd based systems' do
+            let(:facts) { super().merge(service_provider: 'systemd') }
+            it do
+              should contain_systemd__dropin_file('puppetserver.service-limits.conf')
+                .with_ensure('present')
+                .with_filename('limits.conf')
+                .with_unit('puppetserver.service')
+                .with_content("[Service]\nLimitNOFILE=32143")
+            end
+          end
+
+          context 'on non systemd based systems' do
```

Comment:
This will now probably fail for some distros

Reply:
Yeah :( Fixed it! :)